tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Contracts;
+using DevHome.Common.Extensions;
+using Microsoft.UI.Xaml;
+using Windows.ApplicationModel;
+using Windows.System;
+
+namespace DevHome.ExtensionLibrary.ViewModels;
+public partial class InstalledExtensionViewModel : ObservableObject
+{
+    [ObservableProperty]
+    private string _displayName;
+
+    [ObservableProperty]
+    private string _extensionUniqueId;
+
+    [ObservableProperty]
+    private bool _hasSettingsProvider;
+
+    [ObservableProperty]
+    private bool _isExtensionEnabled;
+
+    partial void OnIsExtensionEnabledChanging(bool value)
+    {
+        if (IsExtensionEnabled != value)
+        {
+            Task.Run(() =>
+            {
+                var localSettingsService = Application.Current.GetService<ILocalSettingsService>();
+                return localSettingsService.SaveSettingAsync(ExtensionUniqueId + "-ExtensionDisabled", !value);
+            }).Wait();
+        }
+    }
+
+    public InstalledExtensionViewModel(string displayName, string extensionUniqueId, bool hasSettingsProvider)
+    {
+        _displayName = displayName;
+        _extensionUniqueId = extensionUniqueId;
+        _hasSettingsProvider = hasSettingsProvider;
+
+        IsExtensionEnabled = GetIsExtensionEnabled();
+    }
+
+    private bool GetIsExtensionEnabled()
+    {
+        var isDisabled = Task.Run(() =>
+        {
+            var localSettingsService = Application.Current.GetService<ILocalSettingsService>();
+            return localSettingsService.ReadSettingAsync<bool>(ExtensionUniqueId + "-ExtensionDisabled");
+        }).Result;
+        return !isDisabled;
+    }
+
+    [RelayCommand]
+    private void NavigateSettings()
+    {
+        // TODO: nothing implements ISettingsProvider yet, so this will not be called

```

Comment:
NIT: throw NotImplementedException for future devs who might call this thinking it is implemented.

Reply:
I expect it to be implemented on or about tomorrow morning :)